### PR TITLE
Core: log fill progress

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
Some alive signals during a long fill step

## How was this tested?
with 300 Factorio worlds

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

**Comment by black-sliver** [comment]
> Filling the world with **52800** items.

> Current fill step at x/**16134** items placed.

I feel like this is confusing if you don't know that there are multiple fills gonna happen.
So we should either have a message for *each* fill step (where the sum always ends up being the total item count), or we should hide it by default (verbose?).

**Comment by Berserker66** [comment]
I was hoping "current fill step" would indicate there are multiple. Printing for each step is quite verbose, as we have many. I don't like hiding it either. I could add another for remaining_fill, which would almost always at least do it twice, hence making it more obvious there's multiple.

**Comment by black-sliver** [comment]
ap-roller with 3 yamls per run and `-O`, `--spoiler 0` shows an improvement of ~5% after 2500 runs for `soe,v6,spire,subnautica,zillion,timespinner,smz3,sm64ex,sc2wol,sa2b,ror2,rogues_legacy,raft,minecraft,meritous,hk,factorio,dark_souls_3,archipidle,alttp`

**Comment by black-sliver** [comment]
So, my two suggestions are either
put it behind level = verbose or
have it announce what it's filling.

**Comment by Berserker66** [comment]
image
done

**Comment by black-sliver** [comment]
I agree with the first 2 suggestions from el-u. For 3-4, I am not sure how it's formatted elsewhere, but preferably it should be consistent. For 5, I think the current naming is as good as the suggested one, as it fills (some) dungeon items into (some) dungeons.

Lgtm otherwise.

File: Fill.py [hunk 1]
```diff
@@ -38,16 +42,21 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations:
...
-
     swapped_items: typing.Counter[typing.Tuple[int, str, bool]] = Counter()
     reachable_items: typing.Dict[int, typing.Deque[Item]] = {}
     for item in item_pool:
         reachable_items.setdefault(item.player, deque()).append(item)
 
+    # for progress logging
+    total = min(len(item_pool), len(locations))
+    placed = 0
+    name = name if name else "Unknown"
```
[the remark this answers is not in the document]
Comment (Berserker66) [reply Fill.py:58]: True, missed refactor as I originally had it as Optional[str]

**Review by black-sliver — APPROVED** [review]

**Pull request merged**